### PR TITLE
PR

### PR DESCRIPTION
**Pull request by HibaHabash**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,16 +3,14 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "@testing-library/jest-dom": "^5.16.5",
-    "@testing-library/react": "^13.4.0",
-    "@testing-library/user-event": "^13.5.0",
     "@types/jest": "^27.5.2",
     "@types/node": "^16.18.7",
```
Comment (mshakaE) [inline package.json]: why did you install the node types ? and if needed why are they in the list of dependencies ?

File: package.json [hunk 2]
```diff
@@ -3,16 +3,14 @@
...
     "@types/react": "^18.0.26",
```
Comment (mshakaE) [inline package.json]: are you sure you need the types here in this list ?

File: src/components/List/index.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            movies: state.movies.filter(
+                (movies) =>
+                    movies
+                        .toLowerCase()
+                        .indexOf(e.target.value.toLowerCase()) >= 0
+            ),
+        }));
+    };
+
+    const onSearchCallback = useCallback(onSearch, []);
```
Comment (mshakaE) [inline src/components/List/index.tsx]: what is exactly the use of `useCallback` here ?

File: src/components/List/index.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                                className="button"
+                                onClick={() => addFunction(index)}
+                            >
+                                Add
+                            </button>
+                        </div>
+                    ))}
+                </div>
+                <div className="list">
+                    {state.watchingList.map((element, index) => (
```
Comment (mshakaE) [inline src/components/List/index.tsx]: why is there are  two exactly identical  implementation for exactly the same functionality and html

File: src/components/List/style.scss [hunk 5]
```diff
@@ -0,0 +1,75 @@
+.container {
+    background-color: #063970;
+    min-height: 100vh;
```
Comment (mshakaE) [inline src/components/List/style.scss]: I don't think you need to use vh for height

File: src/components/List/style.scss [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    font-size: 20px;
+    padding: 15px 0px 15px 20px;
+    margin-bottom: 50px;
+    outline: none;
+    background: #ffffe6;
+}
+
+.list-container {
+    display: grid;
+    grid-gap: 2em;
```
Comment (mshakaE) [inline src/components/List/style.scss]: what is the purpose of having grid here ?

**Review by mshakaE — COMMENTED** [review]
why did you have to delete the `package-json.lock file` ? how was it created in the first place if you are using yarn

File: package.json (hunk 1 above)
Comment (HibaHabash) [reply package.json]: when creating CRA i used -template typescript , i think this is the default dependencies for it .

File: src/components/List/index.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+         movieList.findIndex(function (index) {
+            return index.id === id;
+         }),
+         1
+      );
+      setState({ ...state });
+   };
+
+   const removeFunction = (id: number) => {
+      movieList.splice(
```
Comment (mshakaE) [inline src/components/List/index.tsx]: this is not allowed you are mutating the state directly

File: src/components/List/index.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+         ]
+      );
+
+      watchingList.splice(
+         watchingList.findIndex(function (index) {
+            return index.id === id;
+         }),
+         1
+      );
+      setState({ ...state });
```
Comment (mshakaE) [inline src/components/List/index.tsx]: what is exactly is this statement supposed to do ??

**Review by mshakaE — CHANGES_REQUESTED** [review]

**Pull request closed without merging**